src/documents/datatypes/date.ts

```diff
@@ -0,0 +1,149 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { isNullish } from '@/src/lib/utils';
+import { $CustomInspect } from '@/src/lib/constants';
+import { DataAPITime, TableCodec, TableDesCtx, TableSerCtx } from '@/src/documents';
+import { $DeserializeForTable, $SerializeForTable } from '@/src/documents/tables/ser-des/constants';
+
+/**
+ * A shorthand function for `new DataAPIDate(date?)`
+ *
+ * If no date is provided, it defaults to the current date.
+ *
+ * @public
+ */
+export const date = (date?: string | Date | DataAPIDateComponents) => new DataAPIDate(date);
+
+/**
+ * Represents the time components that make up a `DataAPIDate`
+ *
+ * @public
+ */
+export interface DataAPIDateComponents {
+  /**
+   * The year of the date
+   */
+  year: number,
+  /**
+   * The month of the date (should be between 1 and 12)
+   */
+  month: number,
+  /**
+   * The day of the month
+   */
+  date: number,
+}
+
+/**
+ * Represents a `date` column for Data API tables.
+ *
+ * You may use the {@link date} function as a shorthand for creating a new `DataAPIDate`.
+ *
+ * See the official DataStax documentation for more information.
+ *
+ * @public
+ */
+export class DataAPIDate implements TableCodec<typeof DataAPIDate> {
+  readonly #date: string;
+
+  /**
+   * Implementation of `$SerializeForTable` for {@link TableCodec}
+   */
+  public [$SerializeForTable](ctx: TableSerCtx) {
+    return ctx.done(this.toString());
+  };
+
+  /**
+   * Implementation of `$DeserializeForTable` for {@link TableCodec}
+   */
+  public static [$DeserializeForTable](_: unknown, value: any, ctx: TableDesCtx) {
+    return ctx.done(new DataAPIDate(value));
+  }
+
+  /**
+   * Creates a new `DataAPIVector` instance from various formats.
+   *
+   * @param input - The input to create the `DataAPIDate` from
+   */
+  public constructor(input?: string | Date | DataAPIDateComponents) {
+    if (typeof input === 'string') {
+      this.#date = input;
+    } else if (input instanceof Date || isNullish(input)) {
+      input ||= new Date();
+      this.#date = `${input.getFullYear().toString().padStart(4, '0')}-${(input.getMonth() + 1).toString().padStart(2, '0')}-${input.getDate().toString().padStart(2, '0')}`;
+    } else {
+      if (input.month < 1 || input.month > 12) {
+        throw new RangeError('Month must be between 1 and 12 (DataAPIDate month is NOT zero-indexed)');
+      }
+      this.#date = `${input.year.toString().padStart(4, '0') ?? '0000'}-${input.month.toString().padStart(2, '0') ?? '00'}-${input.date.toString().padStart(2, '0') ?? '00'}`;

```

Comment:
Is the `?? '0000'` necessary here? `toString()` and `padStart()` can't return a nullish value